ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/BlockingPrintPublisher.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.impl;
+
+import io.confluent.ksql.api.server.KsqlApiException;
+import io.confluent.ksql.parser.tree.PrintTopic;
+import io.confluent.ksql.reactive.BasePublisher;
+import io.confluent.ksql.rest.Errors;
+import io.confluent.ksql.rest.server.resources.streaming.PrintTopicUtil;
+import io.confluent.ksql.rest.server.resources.streaming.RecordFormatter;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.VertxUtils;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.WorkerExecutor;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.utils.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A query publisher that uses an internal blocking queue to store rows for delivery. It's currently
+ * necessary to use a blocking queue as Kafka Streams delivers message in a synchronous fashion with
+ * no back pressure. If the queue was not blocking then if the subscriber was slow the messages
```

Comment:
This comment looks like it's copied from BlockingQueryPublisher, and doesn't apply to PRINT TOPIC. PRINT TOPIC does not get result records from Kafka Streams. Instead your code has full control over the consumer which is producing records. This means that no backpressure mechanism is needed -- if the subscriber is not asking for more rows, then we simply don't need to call poll on the consumer.

Reply:
Thanks for catching that, I've written a proper description for the BlockingPrintPublisher.